### PR TITLE
Templates 2

### PR DESCRIPTION
**Pull request by yann300**
This PR update the grid view and the env providers view:

- Update labels.
- Update UI.
- Add Icons.
- Fix search using the payload.

### CODE REVIEW

File: apps/remix-ide/src/app/providers/environment-explorer.tsx [hunk 1]
```diff
@@ -99,88 +99,100 @@ export class EnvironmentExplorer extends ViewPlugin {
...
-              title={provider.name}
+              title={provider.displayName}
+              logos={provider.logos}
               classList='EECellStyle'
+              payload={provider.description}
               pinned={this.pinnedProviders.includes(provider.name)}
               pinStateCallback={async (pinned: boolean) => {
                 if (pinned) {
                   this.emit('providerPinned', provider.name, provider)
+                  this.call('notification', 'toast', `"${provider.displayName}" has been added to the Network list of the Deploy & Run Transactions module.`)
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/providers/environment-explorer.tsx:116]: It should be 'added to the ~~Network~~ Environment list of the Deploy & Run Transactions ~module~ plugin'

File: apps/remix-ide/src/app/providers/environment-explorer.tsx [hunk 2]
```diff
@@ -99,88 +99,100 @@ export class EnvironmentExplorer extends ViewPlugin {
...
                   return true
                 }
                 const providerName = await this.call('blockchain', 'getProvider')
                 if (providerName !== provider.name) {
                   this.emit('providerUnpinned', provider.name, provider)
+                  this.call('notification', 'toast', `"${provider.displayName}" has been removed from the Network list of the Deploy & Run Transactions module.`)
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/providers/environment-explorer.tsx:122]: here too

File: apps/remix-ide/src/app/providers/environment-explorer.tsx [hunk 3]
```diff
@@ -99,88 +99,100 @@ export class EnvironmentExplorer extends ViewPlugin {
...
-              title={provider.name}
+              title={provider.displayName}
+              logos={provider.logos}
               classList='EECellStyle'
+              payload={provider.description}
               pinned={this.pinnedProviders.includes(provider.name)}
               pinStateCallback={async (pinned: boolean) => {
                 if (pinned) {
                   this.emit('providerPinned', provider.name, provider)
+                  this.call('notification', 'toast', `"${provider.displayName}" has been added to the Network list of the Deploy & Run Transactions module.`)
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/providers/environment-explorer.tsx:149]: here

File: apps/remix-ide/src/app/providers/environment-explorer.tsx [hunk 4]
```diff
@@ -99,88 +99,100 @@ export class EnvironmentExplorer extends ViewPlugin {
...
                   return true
                 }
                 const providerName = await this.call('blockchain', 'getProvider')
                 if (providerName !== provider.name) {
                   this.emit('providerUnpinned', provider.name, provider)
+                  this.call('notification', 'toast', `"${provider.displayName}" has been removed from the Network list of the Deploy & Run Transactions module.`)
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/providers/environment-explorer.tsx:155]: here

File: apps/remix-ide/src/app/providers/environment-explorer.tsx [hunk 5]
```diff
@@ -99,88 +99,100 @@ export class EnvironmentExplorer extends ViewPlugin {
...
-              title={provider.name}
+              title={provider.displayName}
+              logos={provider.logos}
+              payload={provider.description}
               classList='EECellStyle'
               pinned={this.pinnedProviders.includes(provider.name)}
               pinStateCallback={async (pinned: boolean) => {
                 if (pinned) {
                   this.emit('providerPinned', provider.name, provider)
+                  this.call('notification', 'toast', `"${provider.displayName}" has been added to the Network list of the Deploy & Run Transactions module.`)
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/providers/environment-explorer.tsx:181]: here and everywhere similarly

File: apps/remix-ide/src/app/udapp/run-tab.js [hunk 6]
```diff
@@ -129,13 +129,60 @@ export class RunTab extends ViewPlugin {
   async onInitDone() {
     const udapp = this // eslint-disable-line
 
+    const descriptions = {
+      'vm-cancun': 'Deploy to an in-browser virtual machine which run the cancun fork.',
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/udapp/run-tab.js:133]: 'which ~run~ runs'

File: apps/remix-ide/src/app/udapp/run-tab.js [hunk 7]
```diff
@@ -129,13 +129,60 @@ export class RunTab extends ViewPlugin {
...
+      'vm-shanghai': 'Deploy to an in-browser virtual machine which run the cancun fork.',
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/udapp/run-tab.js:134]: here and similarly everywhere

File: apps/remix-ide/src/app/udapp/run-tab.js (hunk 6 above)
Comment (Aniket-Engg) [reply apps/remix-ide/src/app/udapp/run-tab.js:133]: Names should be capitalized like `Cancun`

File: apps/remix-ide/src/app/udapp/run-tab.js [hunk 8]
```diff
@@ -129,13 +129,60 @@ export class RunTab extends ViewPlugin {
...
+      'vm-paris': 'Deploy to an in-browser virtual machine which run the paris fork.',
+      'vm-london': 'Deploy to an in-browser virtual machine which run the london fork.',
+      'vm-berlin': 'Deploy to an in-browser virtual machine which run the berlin fork.',
+      'vm-mainnet-fork': 'Deploy to an in-browser virtual machine which fork locally the ethereum mainnet.',
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/udapp/run-tab.js:138]: `forks` `Ethereum`

File: apps/remix-ide/src/app/udapp/run-tab.js [hunk 9]
```diff
@@ -129,13 +129,60 @@ export class RunTab extends ViewPlugin {
...
+      'vm-sepolia-fork': 'Deploy to an in-browser virtual machine which fork locally the sepolia testnet.',
+      'vm-custom-fork': 'Deploy to an in-browser virtual machine which fork locally a custom network.',
+      'walletconnect': 'Deploy using wallet connect.',
+      'basic-http-provider': 'Deploy to a custom local network.',
+      'hardhat-provider': 'Deploy to a hardhat development nework.',
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/udapp/run-tab.js:143]: `Hardhat` with capital H, and similarly

File: apps/remix-ide/src/app/udapp/run-tab.js [hunk 10]
```diff
@@ -129,13 +129,60 @@ export class RunTab extends ViewPlugin {
...
+      'vm-berlin': ['assets/img/sleepingRemiCroped.webp'],
+      'vm-mainnet-fork': ['assets/img/guitarRemiCroped.webp'],
+      'vm-sepolia-fork': ['assets/img/sleepingRemiCroped.webp'],
+      'vm-custom-fork': ['assets/img/remixLogo.webp'],
+      'foundry-provider': ['assets/img/foundry.png'],
+      'basic-http-provider': ['assets/img/hexagon-remix-greengrey-texture.png']
+    }
+
     const addProvider = async (position, name, displayName, isInjected, isVM, fork = '', dataId = '', title = '') => {
+      console.log(name)
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/udapp/run-tab.js:177]: remove log

**Review by Aniket-Engg — COMMENTED** [review]
Can you make first letter of cancun, paris, berlin etc. capital?

**Review by ryestew — APPROVED** [review]

**Pull request merged**